fails in cxx syntax: safeint_internal::numeric_type< T >::isInt, "Integer type required"

```
module TryingVector;
 tsPrep();

function tsPrep() {
  local tsSinceEpoch: uint64 = 0;
  local debugOctet: uint8 = 0x21;
  YearAndMonthByte(tsSinceEpoch, 2000, debugOctet);
}

function YearAndMonthByte(inout tsSinceEpoch: uint64, Year: uint64, rawTwoNybbleOfMonth: uint8) : string {
    local PriorMonthsesDays: vector<uint64> = [0,31,59,90, 120,151,181,212, 243,273,304,334];
    if (1970 <= Year) {
      tsSinceEpoch = Year * 365 * 86400;
      tsSinceEpoch -= (1970 * 365 * 86400);
      local Months: uint64;
      local leapYears: uint64 = ((Year - 1969) / 4);
      tsSinceEpoch += (leapYears * 86400);
      if ( 9 < rawTwoNybbleOfMonth >> 4) {
        Months = 0;
      }
      else {
        Months = (rawTwoNybbleOfMonth >> 4);
      }	    
      if (rawTwoNybbleOfMonth & 0x0F <= 9) {
        Months += 10 * (rawTwoNybbleOfMonth & 0x0F);
      }
      if (0 == (Year % 4) && (2 < Months)) {
        tsSinceEpoch += 86400;
      }
      tsSinceEpoch += (PriorMonthsesDays[Months - 1] * 86400);
    }
}
```
gets through spicy syntax checks, but fails in cxx syntax:
```
3148 /usr/local/etc/tmp/spicy$ build/bin/spicyc -j ../Issue_212_MonthVector.spicy 
In file included from TryingVector.cc:6:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/libhilti.h:14:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/safe-int.h:5:
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:825:9: error: static_assert failed due to requirement 'safeint_internal::numeric_type<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >::isInt' "Integer type required"
        static_assert(safeint_internal::numeric_type< T >::isInt, "Integer type required");
        ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:858:96: note: in instantiation of template class 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
            isBothLT32Bit = (safeint_internal::int_traits< T >::isLT32Bit && safeint_internal::int_traits< U >::isLT32Bit),
                                                                                               ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:876:55: note: in instantiation of template class 'safeint_internal::type_compare<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
        IntZone_UintLT32_UintLT32 = safeint_internal::type_compare< T,U >::isBothUnsigned && safeint_internal::type_compare< T,U >::isBothLT32Bit,
                                                      ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:1753:19: note: in instantiation of template class 'IntRegion<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
        method = (IntRegion< T,U >::IntZone_UintLT32_UintLT32  ? MultiplicationState_CastUint :
                  ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:6687:33: note: in instantiation of template class 'MultiplicationMethod<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
    MultiplicationHelper< T, U, MultiplicationMethod< T, U >::method >::template MultiplyThrow< E >( (T)rhs, lhs, ret );
                                ^
TryingVector.cc:41:97: note: in instantiation of function template specialization 'operator*<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > >, hilti::rt::integer::detail::SafeIntException>' requested here
        tsSinceEpoch += PriorMonthsesDays[Months - hilti::rt::integer::safe<std::uint64_t>{1u}] * hilti::rt::integer::safe<std::uint64_t>{86400u};
                                                                                                ^
In file included from TryingVector.cc:6:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/libhilti.h:14:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/safe-int.h:5:
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:859:113: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
            isBothLT64Bit = (safeint_internal::int_traits< T >::isLT64Bit && safeint_internal::int_traits< U >::isLT64Bit)
                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:876:55: note: in instantiation of template class 'safeint_internal::type_compare<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
        IntZone_UintLT32_UintLT32 = safeint_internal::type_compare< T,U >::isBothUnsigned && safeint_internal::type_compare< T,U >::isBothLT32Bit,
                                                      ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:1753:19: note: in instantiation of template class 'IntRegion<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
        method = (IntRegion< T,U >::IntZone_UintLT32_UintLT32  ? MultiplicationState_CastUint :
                  ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:6687:33: note: in instantiation of template class 'MultiplicationMethod<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
    MultiplicationHelper< T, U, MultiplicationMethod< T, U >::method >::template MultiplyThrow< E >( (T)rhs, lhs, ret );
                                ^
TryingVector.cc:41:97: note: in instantiation of function template specialization 'operator*<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > >, hilti::rt::integer::detail::SafeIntException>' requested here
        tsSinceEpoch += PriorMonthsesDays[Months - hilti::rt::integer::safe<std::uint64_t>{1u}] * hilti::rt::integer::safe<std::uint64_t>{86400u};
                                                                                                ^
In file included from TryingVector.cc:6:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/libhilti.h:14:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/safe-int.h:5:
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:877:175: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Uint32_UintLT64   = safeint_internal::type_compare< T,U >::isBothUnsigned && safeint_internal::int_traits< T >::is32Bit && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:1753:19: note: in instantiation of template class 'IntRegion<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
        method = (IntRegion< T,U >::IntZone_UintLT32_UintLT32  ? MultiplicationState_CastUint :
                  ^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:6687:33: note: in instantiation of template class 'MultiplicationMethod<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >' requested here
    MultiplicationHelper< T, U, MultiplicationMethod< T, U >::method >::template MultiplyThrow< E >( (T)rhs, lhs, ret );
                                ^
TryingVector.cc:41:97: note: in instantiation of function template specialization 'operator*<unsigned long long, hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > >, hilti::rt::integer::detail::SafeIntException>' requested here
        tsSinceEpoch += PriorMonthsesDays[Months - hilti::rt::integer::safe<std::uint64_t>{1u}] * hilti::rt::integer::safe<std::uint64_t>{86400u};
                                                                                                ^
In file included from TryingVector.cc:6:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/libhilti.h:14:
In file included from /usr/local/etc/tmp/spicy/hilti/include/hilti/rt/safe-int.h:5:
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:878:177: error: no member named 'is32Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_UintLT32_Uint32   = safeint_internal::type_compare< T,U >::isBothUnsigned && safeint_internal::int_traits< T >::isLT32Bit && safeint_internal::int_traits< U >::is32Bit,
                                                                                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:880:178: error: no member named 'is64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_UintLT64_Uint64    = safeint_internal::type_compare< T,U >::isBothUnsigned && safeint_internal::int_traits< T >::isLT64Bit && safeint_internal::int_traits< U >::is64Bit,
                                                                                                                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:883:158: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Uint32_IntLT64    = safeint_internal::int_traits< T >::isUint32 && std::numeric_limits< U >::is_signed && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:884:160: error: no member named 'isInt32' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_UintLT32_Int32    = !std::numeric_limits< T >::is_signed && safeint_internal::int_traits< T >::isLT32Bit && safeint_internal::int_traits< U >::isInt32,
                                                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:885:158: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Uint64_Int        = safeint_internal::int_traits< T >::isUint64 && std::numeric_limits< U >::is_signed && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:886:160: error: no member named 'isInt64' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_UintLT64_Int64    = !std::numeric_limits< T >::is_signed && safeint_internal::int_traits< T >::isLT64Bit && safeint_internal::int_traits< U >::isInt64,
                                                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:887:119: error: no member named 'isInt64' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Uint64_Int64      = safeint_internal::int_traits< T >::isUint64 && safeint_internal::int_traits< U >::isInt64,
                                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:890:173: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int32_IntLT64     = safeint_internal::type_compare< T,U >::isBothSigned && safeint_internal::int_traits< T >::is32Bit && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:891:175: error: no member named 'is32Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_IntLT32_Int32     = safeint_internal::type_compare< T,U >::isBothSigned && safeint_internal::int_traits< T >::isLT32Bit && safeint_internal::int_traits< U >::is32Bit,
                                                                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:892:173: error: no member named 'isInt64' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int64_Int64       = safeint_internal::type_compare< T,U >::isBothSigned && safeint_internal::int_traits< T >::isInt64 && safeint_internal::int_traits< U >::isInt64,
                                                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:893:173: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int64_Int         = safeint_internal::type_compare< T,U >::isBothSigned && safeint_internal::int_traits< T >::is64Bit && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:894:175: error: no member named 'is64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_IntLT64_Int64     = safeint_internal::type_compare< T,U >::isBothSigned && safeint_internal::int_traits< T >::isLT64Bit && safeint_internal::int_traits< U >::is64Bit,
                                                                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:897:158: error: no member named 'isLT32Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int32_UintLT32    = safeint_internal::int_traits< T >::isInt32 && !std::numeric_limits< U >::is_signed && safeint_internal::int_traits< U >::isLT32Bit,
                                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:898:159: error: no member named 'isUint32' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_IntLT64_Uint32    = std::numeric_limits< T >::is_signed && safeint_internal::int_traits< T >::isLT64Bit && safeint_internal::int_traits< U >::isUint32,
                                                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:899:158: error: no member named 'isLT64Bit' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int64_UintLT64    = safeint_internal::int_traits< T >::isInt64 && !std::numeric_limits< U >::is_signed && safeint_internal::int_traits< U >::isLT64Bit,
                                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:900:111: error: no member named 'isUint64' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int_Uint64        = std::numeric_limits< T >::is_signed && safeint_internal::int_traits< U >::isUint64 && safeint_internal::int_traits< T >::isLT64Bit,
                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/usr/local/etc/tmp/spicy/hilti/include/hilti/3rdparty/SafeInt/SafeInt.hpp:901:118: error: no member named 'isUint64' in 'safeint_internal::int_traits<hilti::rt::vector::detail::AssignProxy<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException>, std::__1::allocator<SafeInt<unsigned long long, hilti::rt::integer::detail::SafeIntException> > > >'
        IntZone_Int64_Uint64      = safeint_internal::int_traits< T >::isInt64 && safeint_internal::int_traits< U >::isUint64
                                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
20 errors generated.
[error] jit: failed to execute compilation action.
[error] jit: failed to compile C++ code unit TryingVector to bitcode
[error] spicyc: JIT compilation failed
3149 /usr/local/etc/tmp/spicy$
```

--- a/hilti/include/rt/types/vector.h
+++ b/hilti/include/rt/types/vector.h
@@ -150,7 +150,12 @@ public:
         return V::data()[i];
     }
 
-    auto operator[](uint64_t i) &;
+    T& operator[](uint64_t i) & {
+        if ( i >= V::size() )
+            V::resize(i + 1);
+
+        return V::data()[i];
+    }
 
     Vector operator+(const Vector& other) const {
         Vector v(*this);
@@ -163,58 +168,6 @@ public:
         return *this;
     }
 };
-
-namespace vector::detail {
-
-template<typename T, typename Allocator>
-class AssignProxy {
-public:
-    using V = std::vector<T, Allocator>;
-
-    AssignProxy(V* v, uint64_t i) : _v(v), _i(i) {}
-
-    AssignProxy& operator=(T t) {
-        _v->resize(std::max(static_cast<uint64_t>(_v->size()), _i + 1));
-        get() = std::move(t);
-        return *this;
-    }
-
-    operator T() const { return get(); }
-
-    bool operator==(const T& t) const { return get() == t; }
-    bool operator!=(const T& t) const { return get() != t; }
-
-    friend std::ostream& operator<<(std::ostream& out, const AssignProxy& x) {
-        out << x.get();
-        return out;
-    }
-
-private:
-    // Get a mutable reference to the wrapped value.
-    //
-    // This method is `const` so we can use it in both `const` and non-`const`
-    // contexts, but `private` since it is not safe (it would e.g., allow
-    // mutating elements of `const` vectors).
-    T& get() const {
-        if ( _i >= _v->size() )
-            throw IndexError(fmt("vector index %" PRIu64 " out of range", _i));
-
-        // Forward to the containers subscript operator.
-        //
-        // `AssignProxy`s are only created for non-const `Vector`s so these casts are safe.
-        return const_cast<T&>(const_cast<const V&>(*_v)[_i]);
-    }
-
-    V* _v;
-    uint64_t _i;
-};
-
-} // namespace vector::detail
-
-template<typename T, typename Allocator>
-inline auto Vector<T, Allocator>::operator[](uint64_t i) & {
-    return hilti::rt::vector::detail::AssignProxy(this, i);
-}
 
 namespace vector {
 /** Place-holder type for an empty vector that doesn't have a known element type. */
@@ -250,11 +203,6 @@ inline std::string to_string(const Vector<T, Allocator>& x, adl::tag /*unused*/)
 template<typename T, typename Allocator>
 inline std::string to_string(const std::vector<T, Allocator>& x, adl::tag /*unused*/) {
     return to_string(static_cast<const Vector<T, Allocator>&>(x), adl::tag{});
-}
-
-template<typename T, typename Allocator>
-inline std::string to_string(const vector::detail::AssignProxy<T, Allocator>& x, adl::tag /*unused*/) {
-    return hilti::rt::to_string(T(x));
 }
 
 inline std::string to_string(const vector::Empty& /* x */, adl::tag /*unused*/) { return "[]"; }

--- a/hilti/src/rt/tests/vector.cc
+++ b/hilti/src/rt/tests/vector.cc
@@ -37,31 +37,15 @@ TEST_CASE("concat") {
 }
 
 TEST_CASE("subscript") {
-    // Size check in subscript access is only performed eagerly if the
-    // access yields an rvalue. If it yields an lvalue the check is
-    // performed lazily and only at the point where the value is accessed.
+    CHECK_THROWS_WITH_AS(Vector<int>()[47], "vector index 47 out of range", const IndexError&);
 
-    SUBCASE("lvalue-semantics") {
-        Vector<int> xs;
+    Vector<int> xs;
+    REQUIRE_EQ(xs.size(), 0u);
+    CHECK_EQ(xs[3], int());
+    CHECK_EQ(xs.size(), 4u);
 
-        // No size check since no access to yielded value.
-        CHECK_NOTHROW(xs[47]);
-
-        // Access to the yielded value triggers a size check.
-        CHECK_THROWS_WITH_AS((void)int(xs[47]), "vector index 47 out of range", const IndexError&);
-
-        // Assigning to the yielded value potentially resizes the vector.
-        xs[47] = 11;
-        CHECK_EQ(xs.size(), 47 + 1); // Vector resized.
-        CHECK_EQ(xs[12], int());     // Added elements default-initialzed.
-        CHECK_EQ(xs[47], 11);        // Assignment is to added value.
-    }
-
-    SUBCASE("rvalue-semantics") {
-        const Vector<int> xs;
-        CHECK_THROWS_WITH_AS(xs[47], "vector index 47 out of range", const IndexError&);
-        CHECK_THROWS_WITH_AS(Vector<int>()[47], "vector index 47 out of range", const IndexError&);
-    }
+    const auto ys = xs;
+    CHECK_THROWS_WITH_AS(ys[47], "vector index 47 out of range", const IndexError&);
 }
 
 TEST_SUITE_END();

--- a/tests/hilti/types/vector/ops.hlt
+++ b/tests/hilti/types/vector/ops.hlt
@@ -16,7 +16,6 @@ assert v[1] != "X";
 v[5] = "e";
 assert v[5] == "e";
 assert v[4] == "";
-assert-exception hilti::print(v[100]); # Need to use the result to trigger exeception;
 
 v.push_back("f");
 v.push_back("g");

